runtime/foucoco/src/lib.rs

```diff
@@ -893,6 +900,517 @@ parameter_types! {
+	ConsumerRemaining,
+	/// There are no providers so the account cannot be created.
+	NoProviders,
+	/// There are too many consumers so the account cannot be created.
+	TooManyConsumers,
+	/// An error to do with tokens.
+	Token(PalletAssetTokenErr),
+	/// An arithmetic error.
+	Arithmetic(PalletAssetArithmeticErr),
+	//unknown error
+	Unknown,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, Decode, MaxEncodedLen)]
+pub enum PalletAssetArithmeticErr {
+	/// Underflow.
+	Underflow,
+	/// Overflow.
+	Overflow,
+	/// Division by zero.
+	DivisionByZero,
+	//unknown error
+	Unknown,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Encode, Decode, MaxEncodedLen)]
+pub enum PalletAssetTokenErr {
+	/// Funds are unavailable.
+	NoFunds,
+	/// Account that must exist would die.
+	WouldDie,
+	/// Account cannot exist with the funds that would be given.
+	BelowMinimum,
+	/// Account cannot be created.
+	CannotCreate,
+	/// The asset in question is unknown.
+	UnknownAsset,
+	/// Funds exist but are frozen.
+	Frozen,
+	/// Operation is not supported by the asset.
+	Unsupported,
+	//unknown error
+	Unknown,
+}
+
+impl From<DispatchError> for ChainExnensionErr {
+	fn from(e: DispatchError) -> Self {
+		match e {
+			DispatchError::Other(_) => ChainExnensionErr::Other,
+			DispatchError::CannotLookup => ChainExnensionErr::CannotLookup,
+			DispatchError::BadOrigin => ChainExnensionErr::BadOrigin,
+			DispatchError::Module(_) => ChainExnensionErr::Module,
+			DispatchError::ConsumerRemaining => ChainExnensionErr::ConsumerRemaining,
+			DispatchError::NoProviders => ChainExnensionErr::NoProviders,
+			DispatchError::TooManyConsumers => ChainExnensionErr::TooManyConsumers,
+			DispatchError::Token(token_err) =>
+				ChainExnensionErr::Token(PalletAssetTokenErr::from(token_err)),
+			DispatchError::Arithmetic(arithmetic_error) =>
+				ChainExnensionErr::Arithmetic(PalletAssetArithmeticErr::from(arithmetic_error)),
+			_ => ChainExnensionErr::Unknown,
+		}
+	}
+}
+
+impl From<ArithmeticError> for PalletAssetArithmeticErr {
+	fn from(e: ArithmeticError) -> Self {
+		match e {
+			ArithmeticError::Underflow => PalletAssetArithmeticErr::Underflow,
+			ArithmeticError::Overflow => PalletAssetArithmeticErr::Overflow,
+			ArithmeticError::DivisionByZero => PalletAssetArithmeticErr::DivisionByZero,
+		}
+	}
+}
+
+impl From<TokenError> for PalletAssetTokenErr {
+	fn from(e: TokenError) -> Self {
+		match e {
+			TokenError::NoFunds => PalletAssetTokenErr::NoFunds,
+			TokenError::WouldDie => PalletAssetTokenErr::WouldDie,
+			TokenError::BelowMinimum => PalletAssetTokenErr::BelowMinimum,
+			TokenError::CannotCreate => PalletAssetTokenErr::CannotCreate,
+			TokenError::UnknownAsset => PalletAssetTokenErr::UnknownAsset,
+			TokenError::Frozen => PalletAssetTokenErr::Frozen,
+			TokenError::Unsupported => PalletAssetTokenErr::Unsupported,
+		}
+	}
+}
+
+fn try_from(type_id: u8, code: [u8; 12], issuer: [u8; 32]) -> Result<CurrencyId, ()> {
```

Comment:
Can we move all this stuff to a separate file maybe? Maybe to the `common` directory because I assume we will need to have the same code also in the other runtimes.

Reply:
Done